src/main/java/graphql/parser/Parser.java

```diff
@@ -7,18 +7,24 @@
 import org.antlr.v4.runtime.BailErrorStrategy;
 import org.antlr.v4.runtime.CharStreams;
 import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.IntStream;
 import org.antlr.v4.runtime.Token;
 import org.antlr.v4.runtime.atn.PredictionMode;
 import org.antlr.v4.runtime.misc.ParseCancellationException;
 
 import java.util.List;
+import java.util.Optional;
 
 @Internal
 public class Parser {
 
     public Document parseDocument(String input) {
+        return parseDocument(input, null);
+    }
+
+    public Document parseDocument(String input, String sourceName) {
 
-        GraphqlLexer lexer = new GraphqlLexer(CharStreams.fromString(input));
+        GraphqlLexer lexer = new GraphqlLexer(CharStreams.fromString(input, Optional.ofNullable(sourceName).orElse(IntStream.UNKNOWN_SOURCE_NAME)));
```

Comment:
Can we please change the default from IntStream.UNKNOWN_SOURCE_NAME (which equals the string "<unknown>") and change it to null.

My reasoning is for non SDL documents.  Today a graphql query will get errors without this value and with this PR it will say "<unknown>"

Most Object -> JSON conversions will leave out null columns, so null will mean the JSON output doesn't change for client queries.

Its a change of least impact this way to existing clients

Reply:
Yeah my point is not at the parser level but within the SourceLocation object.

The SourceLocation is sent back in the error objects.  Today there is no sourceName attribute.  Your PR adds once.

So the JSON today will be

     errors : {
        message : "x y z",
        locations : [
            {  line : 10, column : 9  }
       ],
     }

with your change a bad client query (as opposed to SDL) will send back

     errors : {
        message : "x y z",
        locations : [
            { sourceName : "<unknown>", line : 10, column : 9  }
       ],
     }

 But the client knows the source - the query it just sent.

Can we default it to NULL IF the source name is the magic constant

eg

    String sourceName = startToken.getTokenSource().getSourceName();
    if (IntStream.UNKNOWN_SOURCE_NAME.equals(sourceName)) {
       sourceName = null;
    }
     return new SourceLocation(startToken.getLine(), startToken.getCharPositionInLine() + 1, sourceName);

Reply:
Remember the same grammar and parser is used for both queries and SDL definitions

Reply:
Ah, makes sense.

Didn't get that from your first comment. I was focusing on the parser instantiation code.

I'll update the PR.